### PR TITLE
VOTE SLEP 17

### PR DESCRIPTION
**Pull request by amueller**
This PR is for us to discuss and collect votes for [SLEP017 - Clone Override Protocol with __sklearn_clone__](https://scikit-learn-enhancement-proposals.readthedocs.io/en/latest/slep017/proposal.html). The current implementation is available at https://github.com/scikit-learn/scikit-learn/pull/24568. Note that this vote is for the API and the implementation can be adjusted.

According to our [governance model](https://scikit-learn.org/stable/governance.html#decision-making-process), the vote will be open for a month (till Dec 1st), and the motion is accepted if 2/3 of the cast votes are in favor.

@scikit-learn/core-devs

### CODE REVIEW

**Review by adrinjalali — APPROVED** [review]

**Review by agramfort — APPROVED** [review]

**Review by rth — APPROVED** [review]

**Review by TomDLT — APPROVED** [review]

**Comment by glemaitre** [comment]
A question before voting.

The SLEP states that we will use the estimator checks to ensure that the fundamental properties of `clone` are preserved. Currently, we don't have such a check. I am wondering if the SLEP should define "the contract" that we will be checking in `check_estimator`?

**Comment by adrinjalali** [comment]
>  am wondering if the SLEP should define "the contract" that we will be checking in check_estimator?

I rather have that discussion concretely in the PR where we write those tests instead. Having that discussion here would add another 6 months to this SLEP's discussions.

**Review by glemaitre — APPROVED** [review]
Fine with me then.

+1

File: slep017/proposal.rst [hunk 1]
```diff
@@ -6,7 +6,7 @@ Clone Override Protocol with ``__sklearn_clone__``
 :Status: Draft
 :Type: Standards Track
 :Created: 2022-03-19
-:Resolution: (required for Accepted | Rejected | Withdrawn)
+:Resolution: Accepted
```
Comment (jjerphan) [inline slep017/proposal.rst:9]: Based on [the description of the header preambule on SLEP000](https://scikit-learn-enhancement-proposals--74.org.readthedocs.build/en/74/slep000/proposal.html#header-preamble), it seems the header of this SLEP must be changed to this one (I am not sure about the format of `scikit-learn-Version`) :

```rst
:Author: Joel Nothman
:Status: Accepted
:Type: Standards Track
:Created: 2022-03-19
:scikit-learn-Version: 1.2.0
:Resolution: https://github.com/scikit-learn/enhancement_proposals/pull/79
```

What do you think?

**Review by lorentzenchr — APPROVED** [review]

**Review by Micky774 — APPROVED** [review]

File: slep017/proposal.rst (hunk 1 above)
Comment (jjerphan) [reply slep017/proposal.rst:9]: I have opened https://github.com/amueller/enhancement_proposals/pull/1 accordingly.

**Review by jnothman — APPROVED** [review]

**Review by thomasjpfan — APPROVED** [review]

File: slep017/proposal.rst [hunk 2]
```diff
@@ -3,10 +3,11 @@ Clone Override Protocol with ``__sklearn_clone__``
 ==================================================
 
 :Author: Joel Nothman
-:Status: Draft
+:Status: Accepted
 :Type: Standards Track
 :Created: 2022-03-19
-:Resolution: (required for Accepted | Rejected | Withdrawn)
+:scikit-learn-Version: 1.2.0
```
Comment (adrinjalali) [inline slep017/proposal.rst:9]: pretty sure we can't get it in by 1.2 release :D

Comment (jjerphan) [reply slep017/proposal.rst:9]: Almost surely, yes. :smile: 

I would turn it to 1.2.1 when 1.2.0 is released.
```suggestion
:scikit-learn-Version: 1.2.1
```

What do you think?

File: slep017/proposal.rst (hunk 1 above)
Comment (lorentzenchr) [reply slep017/proposal.rst]: I think that it is enough to add the version number once it is implemented. We do not vote for a release version.

File: slep017/proposal.rst (hunk 2 above)
Comment (adrinjalali) [reply slep017/proposal.rst:9]: such a feature can't go in a minor release, it'd be more like 1.3

Comment (jjerphan) [reply slep017/proposal.rst:9]: In that case:
```suggestion
:scikit-learn-Version: 1.3.0
```

**Comment by jnothman** [comment]
Merging, because I can.

**Pull request merged**